hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/util/FilterGenVisitor.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.hive.util;
+
+import org.apache.hudi.hive.expression.Expression;
+import org.apache.hudi.hive.expression.ExpressionVisitor;
+import org.apache.hudi.hive.expression.LeafExpression;
+
+import java.util.Locale;
+
+public class FilterGenVisitor implements ExpressionVisitor<String> {
+
+  private String makeBinaryOperatorString(String left,  Expression.Operator operator, String right) {
+    return String.format("%s %s %s", left, operator.sqlOperator, right);
+  }
+
+  private String quoteStringLiteral(String value) {
+    if (!value.contains("\"")) {
+      return "\"" + value + "\"";
+    } else if (!value.contains("'")) {
+      return "'" + value + "'";
+    } else {
+      throw new UnsupportedOperationException("Cannot pushdown filters if \" and ' both exist");
+    }
+  }
+
+  @Override
+  public String visitAnd(Expression left, Expression right) {
+    String leftResult = left.accept(this);
+    String rightResult = right.accept(this);
+
+    if (leftResult.isEmpty() && rightResult.isEmpty()) {
```

Comment:
You can simplify these conditional if you chain them as if (...) ... else if (...)